src/Compiler/TaskManager.h

```diff
@@ -109,6 +114,7 @@ class TaskManager : public QObject {
   QMap<uint, Task *> m_tasks;
   QVector<Task *> m_runStack;
   QMap<Task *, QVector<Task *>> m_rollBack;
+  std::optional<int> m_taskCount{std::nullopt};
```

Comment:
Can we use anything from Qt here for optional value? I see that this is likely the only std we are using in this primarily Qt file.

Reply:
I don't see the problem with using anything from the std. This is even better.

Reply:
This is the question for discussion but I think we should use std there it is possible.

Reply:
I am looking to see if we can separate out Qt vs Std code where we can for testing reasons down the line.